### PR TITLE
EN-11909 Do spoofing

### PR DESCRIPTION
**Pull request by rozap**

### CODE REVIEW

File: es6-lib/errors.js [hunk 1]
```diff
@@ -306,7 +306,11 @@ class CleanupError extends UpstreamError {
     return 'Encountered an error, but encountered another error rolling back';
   }
 }
-
+class AuthenticationError extends UpstreamError {
+  static template() {
+    return 'Encountered an while authenticating';
```
Comment (catstavi) [inline es6-lib/errors.js]: "an error"?

File: es6-lib/upstream/auth.js [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+          form: {
+            username: `${email} ${username}`,
+            password
+          },
+          headers: {
+            'X-Socrata-Host': this.host
+          }
+        }, (err, response, body) => {
+          if(err) onComplete(new AuthenticationError(response.statusCode, body));
+          this.cookie = response.headers['set-cookie'];
```
Comment (catstavi) [inline es6-lib/upstream/auth.js]: is this case insensitive?

File: es6-lib/upstream/auth.js [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+          user: username,
+          pass: password
+        },
+        timeout,
+        headers: {
+          'X-Socrata-Host': this.host
+        },
+        json: true
+      }, (err, response, {email}) => {
+        if(err)
```
Comment (catstavi) [inline es6-lib/upstream/auth.js]: should this stop, if err?

Comment (rozap) [inline es6-lib/upstream/auth.js]: oh wow..oops

File: es6-lib/upstream/auth.js (hunk 2 above)
Comment (rozap) [inline es6-lib/upstream/auth.js]: yep

File: es6-test/services/mock-core.js [hunk 4]
```diff
@@ -90,6 +89,13 @@ class CoreMock {
     app.use(/\/views$/, enforceUA, bodyParser.json());
     app.use(/.*columns$/, enforceUA, bodyParser.json());
 
+    app.post('/authenticate', (req, res) => {
+      this._history.push(req);
+      return res.status(200)
+        .header('set-cookie', 'monster')
```
Comment (catstavi) [inline es6-test/services/mock-core.js]: 🍪

**Comment by rozap** [comment]
yep - tried this locally and things were happy

**Pull request merged**